### PR TITLE
Add exception when trying to load from a directory (or a file name with extension)

### LINKED ISSUE #4610: Bug: `LOAD FROM` a directory causes seg fault
### Kùzu version

master

### What operating system are you using?

MAC M1

### What happened?

Load from a directory causes seg fault/exception in kuzu:
```
kuzu> load from '/tmp' return *;
Error: basic_string
```

### Are there known steps to reproduce?

Run the following query in shell:
```
kuzu> load from '/tmp' return *;
Error: basic_string
```

### PR DESCRIPTION
**Pull request by SterlingT3485**
# Description

Added a fileExists function to check if a path is both 1.a file 2. existed . 

This check is performed in Binder::getScanFunction for all FileType except for UNKNOWN 

Fixes #4610 

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: src/binder/bind/bind_file_scan.cpp [hunk 1]
```diff
@@ -24,6 +24,10 @@ namespace binder {
 FileTypeInfo bindSingleFileType(main::ClientContext* context, const std::string& filePath) {
     std::filesystem::path fileName(filePath);
     auto extension = context->getVFSUnsafe()->getFileExtension(fileName);
+    if (extension.empty()) {
```
Comment (acquamarin) [inline src/binder/bind/bind_file_scan.cpp:27]: I think the extension name might be empty if you are scanning from the delta/iceberg. So we shouldn't blindly error when the extension is an empty string.
Can you check how duckdb handle this?

Comment (SterlingT3485) [reply src/binder/bind/bind_file_scan.cpp:27]: let me check this

Comment (SterlingT3485) [reply src/binder/bind/bind_file_scan.cpp]: Fixed this

File: src/binder/binder.cpp [hunk 2]
```diff
@@ -205,6 +206,17 @@ function::TableFunction Binder::getScanFunction(FileTypeInfo typeInfo, const Rea
     std::vector<LogicalType> inputTypes;
     inputTypes.push_back(LogicalType::STRING());
     auto functions = clientContext->getCatalog()->getFunctions(clientContext->getTx());
+    // If we defined a certain FileType, we have to ensure the path is a file, not something else
```
[the remark this answers is not in the document]
Comment (SterlingT3485) [reply src/binder/binder.cpp]: moved

**Pull request merged**